narwhal/types/src/consensus.rs

```diff
@@ -167,97 +229,98 @@ impl CommittedSubDagShell {
     }
 }
 
-/// Shutdown token dropped when a task is properly shut down.
-pub type ShutdownToken = mpsc::Sender<()>;
+impl CompressedCommittedSubDagAPI for CompressedCommittedSubDagV2 {
+    fn certificates(&self) -> Vec<CertificateDigest> {
+        self.certificates.clone()
+    }
 
-/// Convenience type to propagate store errors.
-pub type StoreResult<T> = Result<T, TypedStoreError>;
+    fn leader(&self) -> CertificateDigest {
+        self.leader
+    }
 
-/// The persistent storage of the sequencer.
-pub struct ConsensusStore {
-    /// The latest committed round of each validator.
-    last_committed: DBMap<AuthorityIdentifier, Round>,
-    /// The global consensus sequence.
-    committed_sub_dags_by_index: DBMap<SequenceNumber, CommittedSubDagShell>,
+    fn leader_round(&self) -> Round {
+        self.leader_round
+    }
+
+    fn sub_dag_index(&self) -> SequenceNumber {
+        self.sub_dag_index
+    }
+
+    fn reputation_score(&self) -> ReputationScores {
+        self.reputation_score.clone()
+    }
+
+    fn commit_timestamp(&self) -> TimestampMs {
+        self.commit_timestamp
+    }
 }
 
-impl ConsensusStore {
-    /// Create a new consensus store structure by using already loaded maps.
-    pub fn new(
-        last_committed: DBMap<AuthorityIdentifier, Round>,
-        sequence: DBMap<SequenceNumber, CommittedSubDagShell>,
-    ) -> Self {
-        Self {
-            last_committed,
-            committed_sub_dags_by_index: sequence,
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[enum_dispatch(CompressedCommittedSubDagAPI)]
+pub enum CompressedCommittedSubDag {
```

Comment:
Practically I couldn't touch the `CommittedSubDagShell`, thus I came up with a new naming convention that can be used from now on.

Reply:
What about `ConsensusCommit` ?

Reply:
Sounds great!